File: JS_Cluster/Apps/Main_Server/src/Models/Trip.ts
import { RowDataPacket } from "mysql2";

export enum TripStatus {
    CANCELLED = "cancelled",
    PENDING = "pending",
    COMPLETE = "complete",
    IN_PROGRESS = "in progress"
}

export interface Trip extends RowDataPacket {
    id: string,
    bike_id: string,
    customer_id: string,
    trip_status: TripStatus
    reservation_expiry: number,
    trip_start_date?: number,
    trip_end_date?: number,
    trip_start_long: number,
    trip_start_lat: number,
    trip_end_long?: number,
    trip_end_lat?: number,
    trip_secret?: string,
}

File: GoScootAndroid/app/src/main/java/com/example/goscootandroid/Models/Domains/Trip.kt
package com.example.goscootandroid.Models.Domains

import android.annotation.SuppressLint
import kotlinx.serialization.Serializable

@Serializable
enum class TripStatus(val value: String) {
    CANCELLED("Cancelled"),
    PENDING("Pending"),
    COMPLETE("Complete"),
    IN_PROGRESS("In Progress")
}
@SuppressLint("UnsafeOptInUsageError")
@Serializable
data class Trip (
    val id: String,
    val bike: Bike,
    val hub: BikeHub,
    val customer_id: String,
    val trip_status: TripStatus,
    val reservation_expiry: Long,
    val reservation_date: Long,
    val trip_start_date: Long?,
    val trip_end_date: Long?,
    val trip_end_long: Double?,
    val trip_end_lat: Double?,
    val trip_secret: String?,
    val price: Int?,
    val isPaid: Boolean?,
)